src/main/java/crawlercommons/sitemaps/sax/XMLHandler.java

```diff
@@ -72,7 +72,20 @@
     public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
         // flush any unclosed or missing URL element
         if (loc.length() > 0 && ("loc".equals(qName) || "url".equals(qName))) {
-            maybeAddSiteMapUrl();
+            // check whether loc isn't white space only
+            for (int i = 0; i < loc.length(); i++) {
+                if (!Character.isWhitespace(loc.charAt(i))) {
+                    maybeAddSiteMapUrl();
+                    return;
+                }
+            }
+            loc = new StringBuilder();
```

Comment:
A URL is added in both cases, as far as we have more than whitespace buffered in `loc`. But a call of maybeAddSiteMapUrl() turned out to be expensive. I didn't want to move the test for whitespace-only strings to this function to keep the difference between empty `<loc></loc>` and only a opening `<loc>` or `<url>`.